src/Core/View.js

```diff
@@ -920,8 +926,12 @@ class View extends THREE.EventDispatcher {
 
                     precision = CRS.isMetricUnit(texture.features.crs) ? precisions.M : precisions.D;
 
-                    result[materialLayer.id] = result[materialLayer.id].concat(
-                        FeaturesUtils.filterFeaturesUnderCoordinate(coordinates, texture.features, precision));
+                    const featuresUnderCoor = FeaturesUtils.filterFeaturesUnderCoordinate(coordinates, texture.features, precision);
+                    featuresUnderCoor.forEach((feature) => {
+                        if (!result[materialLayer.id].find(f => f.geometry === feature.geometry)) {
```

Comment:
Shouldn't we add an id to `Feature` (that  would be generated) to simplify such comparisons ? (can be done in another PR)

Reply:
I thought about it as well, but not sure it is will easy to set up ans usefull for anything else.